In [5]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split

df=pd.read_csv('data/Loan_default.csv')
print(df.shape)

(255347, 18)


In [7]:
#separate features and target
x=df.drop(['Default','LoanID'], axis=1)
y=df['Default']

print(x.shape)
print(x.dtypes)

(255347, 16)
Age                 int64
Income              int64
LoanAmount          int64
CreditScore         int64
MonthsEmployed      int64
NumCreditLines      int64
InterestRate      float64
LoanTerm            int64
DTIRatio          float64
Education             str
EmploymentType        str
MaritalStatus         str
HasMortgage           str
HasDependents         str
LoanPurpose           str
HasCoSigner           str
dtype: object


In [9]:
#encode categorical columns. text into numbers
le = LabelEncoder()

cat_cols=['Education', 'EmploymentType', 'MaritalStatus', 'HasMortgage','HasDependents','LoanPurpose','HasCoSigner']

for col in cat_cols:
    x[col]=le.fit_transform(x[col])

print(x.dtypes)

Age                 int64
Income              int64
LoanAmount          int64
CreditScore         int64
MonthsEmployed      int64
NumCreditLines      int64
InterestRate      float64
LoanTerm            int64
DTIRatio          float64
Education           int64
EmploymentType      int64
MaritalStatus       int64
HasMortgage         int64
HasDependents       int64
LoanPurpose         int64
HasCoSigner         int64
dtype: object


What LabelEncoder did:
For example EmploymentType which had values like Full-time, Part-time, Self-employed, Unemployed — it converted them to 0, 1, 2, 3.

In [10]:
#split into train and test sets
x_train, x_test, y_train, y_test = train_test_split(
    x, y, test_size=0.2, random_state=42, stratify=y
)

print('Training set:', x_train.shape)
print('Test set:', x_test.shape)
print('\nTraining target distribution:')
print(y_train.value_counts(normalize=True))

Training set: (204277, 16)
Test set: (51070, 16)

Training target distribution:
Default
0    0.883873
1    0.116127
Name: proportion, dtype: float64


test_size=0.2 — 80% of data goes to training, 20% to testing.

stratify=class imbalance 88% non-default, 12% default. random split might be unrepresentative. starify guarantee both train and test sets maintain the same 88/12 ratio as the original data